src/frontend/ocamlmerlin/ocamlmerlin.c

```diff
@@ -521,30 +522,84 @@ static void compute_merlinpath(char merlin_path[PATHSZ], const char *argv0, stru
 }
 
 #ifdef _WIN32
+
+/* May return NULL */
+char * retrieve_user_sid_string()
+{
+  char * usidstr = NULL;
+
+  void * process_token = NULL;
+  if ( ! OpenProcessToken( GetCurrentProcess(), TOKEN_QUERY, &process_token ) )
+    return usidstr;
+
+  unsigned long sid_buffer_size = 0;
+  if ( ! GetTokenInformation(process_token, TokenUser, NULL, 0, &sid_buffer_size ) &&
+        ( GetLastError() != ERROR_INSUFFICIENT_BUFFER ) )
+  {
+    CloseHandle( process_token );
+    process_token = NULL;
+    return usidstr;
```

Comment:
```suggestion
    return NULL;
```
(this isn't C++ - the assignment of `NULL` should be eliminated by the optimiser anyway 🙂)